### ISSUE TITLE
amp-iframe allow-same-orign sandbox.

### ISSUE THREAD

**Issue by Nemo64** [ampproject/amphtml#1266]
I just tried implementing Disqus on an amp page using an `<amp-iframe>` without much luck. I found the only way the discus script works correctly is to use `allow-same-origin`. However amp does not allow the src to be on the same origin when that is set. (why is that anyways?)

I found that using `allow-Same-origin` (capital S) can work around this restriction but that's probably not intended.

**Comment by dvoytenko** [comment]
@Nemo64 the AMP documents meant to be portable - i.e. shown under the different origin such as cdn.ampproject.org. This is where the requirement comes from.

I'm not too familiar with Disqus. Do you know why it needs `allow-same-origin`?

**Comment by Nemo64** [comment]
Well, basically `allow-same-origin` is a workaround for this problem: https://code.google.com/p/chromium/issues/detail?id=541221

However (other tan documented in the issue) this problem also exists in firefox so there might be more involved. I already asked on [stackoverflow](http://stackoverflow.com/questions/34520908/disquss-on-an-amp-page-inside-an-amp-iframe/34528757#34528757) so there is more information on my specific issue.

The portability argument is good but shouldn't isn't an url with domain sufficient then?

**Comment by cramforce** [comment]
@Nemo64 Could you add a failing example file? Maybe on jsbin?

You are allowed to use `allow-same-origin` if the iframe isn't actually on the same origin. This is still useful since the restriction is transitive.

**Comment by Nemo64** [comment]
@cramforce I think i only now fully understood what the `allow-same-origin` does. [w3schools has an example](http://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_iframe_sandbox_origin)  dedicated to it where the error message in chrome does give a better hint. `Origin 'null' is [...] not allowed access.`. So basically, if i omit `allow-same-orign` the sandboxed document will be considered coming from `null`, so it is impossible to make ajax requests because every domain will be a foreign domain, right? That'll, of course, break discus as it does a lot of ajax requests.

So basically i understand why it mustn't be allowed to use `allow-same-origin` when the url is relative, because of the portability of the amp document. But wouldn't it be ok to add it when the url is absolute (contains the domain)?

**Comment by cramforce** [comment]
We don't really make a distinction between relative URLs and same origin URLs.

If you see an error where both parent page and iframe are on localhost, what you can do is to add a few more alternative names to your `/etc/hosts` for localhost and use one of them for the iframes. You just need to make sure that when you deploy it to prod you also have two different domains.

E.g. load the page from `your-site.com` and the iframes from `assets.your-site.com` or something similar.

Does that make sense?

**Comment by Nemo64** [comment]
Yes it does. I need 2 origins for my Iframe stuff to work correctly. I am
already doing this even though it makes the environment much less portable.

Ok, so allow-same-origin works as intended. However there is still the case
that the check for that flag is case insensitive ;)

Malte Ubl <notifications@github.com> schrieb am Fr., 15. Jan. 2016, 15:20:

**Comment by cramforce** [comment]
Are you still seeing the case problem? I believe we fixed it.
